### PR TITLE
Implement Gitlab Releases

### PR DESCRIPTION
**Pull request by cliedeman**

### CODE REVIEW

File: cmd/semantic-release/main.go [hunk 1]
```diff
@@ -57,7 +56,18 @@ func cliHandler(c *cli.Context) error {
...
 	logger.Printf("detected CI: %s\n", ci.Name())
 
-	repo, err := semrel.NewGithubRepository(context.TODO(), conf.GheHost, conf.Slug, conf.Token)
+	var (
+		repo semrel.Repository
+		err  error
+	)
+
+	switch ci.Name() {
+	case "Gitlab":
```
Comment (cliedeman) [inline cmd/semantic-release/main.go:65]: Do we want a provider enum? Seems overkill for this single place

**Comment by cliedeman** [comment]
🎉 it works

Screenshot 2020-04-23 at 22 35 59

@christophwitzko is it expected that it jumps straight to 1.0.0?

File: cmd/semantic-release/main.go (hunk 1 above)
Comment (christophwitzko) [reply cmd/semantic-release/main.go:65]: This is fine for me 👍

File: pkg/semrel/github.go [hunk 2]
```diff
@@ -18,6 +17,7 @@ type GithubRepository struct {
 	repo   string
 	Ctx    context.Context
 	Client *github.Client
+	branch string
```
Comment (christophwitzko) [inline pkg/semrel/github.go:20]: I think we don't need `branch` in this struct.

**Comment by christophwitzko** [comment]
very nice and yes, 1.0.0 is expected 👍

**Comment by cliedeman** [comment]
> very nice and yes, 1.0.0 is expected 👍

Hmmm. Do you know how I can make it start at 0.0.1 instead?

**Comment by christophwitzko** [comment]
I think you could use the `.semrelrc` with
```
{
  "maintainedVersion": "0.0.1"
}
```

**Comment by christophwitzko** [comment]
Thank you for this PR 🎉

**Comment by christophwitzko** [comment]
> I think you could use the `.semrelrc` with
> 
> ```
> {
>   "maintainedVersion": "0.0.1"
> }
> ```

Oh, I just realized that won't work. semantic-release enforces you to start with 1.x.x

**Comment by cliedeman** [comment]
I’ll open up a new issue for that then. Especially for go I would like to stay under v1 and then only release v1 when stable

**Comment by christophwitzko** [comment]
I agree, thank you 👏

**Pull request merged**